carma-messenger-core/traffic_incident/src/traffic_incident_node.cpp

```diff
@@ -61,37 +74,11 @@
    * **/
   bool TrafficIncidentNode::startTrafficBroadcastCallback(cav_srvs::SetTrafficEventRequest& req, cav_srvs::SetTrafficEventResponse& resp)
   {
-    try{
-          //update instance variables with incoming request params
-          traffic_worker_.setMinGap(req.minimum_gap);
-          traffic_worker_.setDownTrack(req.down_track);
-          traffic_worker_.setUpTrack(req.up_track);
-          traffic_worker_.setAdvisorySpeed(req.advisory_speed);
-          ros::Rate loop_rate(10);
-          while (ros::ok())
-          {
-              //construct local mobilityOperation msg
-              cav_msgs::MobilityOperation traffic_mobility_msg = traffic_worker_.mobilityMessageGenerator(traffic_worker_.getPinPoint());
-              
-              if(traffic_worker_.getDownTrack()>0  && traffic_worker_.getUpTrack()>0 && traffic_worker_.getMinGap() > 0) 
-              {
-                //start constantly broadcasting mobilityOperation msg
-                traffic_mobility_operation_pub_.publish(traffic_mobility_msg);
-
-                //return service response true 
-                resp.success=true;
-              }
-
-              ros::spinOnce();
-              loop_rate.sleep();
-          }
-          return true;
-
-        }catch(...){
-              //in case any exception
-              resp.success = false;
-              return false;
-        }    
+      //update instance variables with incoming request params
+      traffic_worker_.setMinGap(req.minimum_gap);
+      traffic_worker_.setDownTrack(req.down_track);
+      traffic_worker_.setUpTrack(req.up_track);
+      traffic_worker_.setAdvisorySpeed(req.advisory_speed);
```

Comment:
You should return true at the end of the service callback.

Reply:
https://github.com/usdot-fhwa-stol/carma-messenger/pull/84/commits/d96950eae103d8a6630d50903cd1175a74af6208